mesonbuild/rewriter.py

```diff
@@ -1077,10 +1157,21 @@ def run(options: argparse.Namespace) -> int:
         if not isinstance(commands, list):
             raise TypeError('Command is not a list')
 
-        for i in commands:
-            if not isinstance(i, object):
+        for i, cmd in enumerate(commands):
+            if not isinstance(cmd, object):
                 raise TypeError('Command is not an object')
-            rewriter.process(i)
+            rewriter.process(cmd)
+
+            if i == len(commands) - 1: # Improves the performance, is not necessary for correctness.
+                continue
```

Comment:
break?

Reply:
It's the same, I would let the submitter follow his taste...